test/upgrade/upgrade_test.go

```diff
@@ -0,0 +1,67 @@
+package upgrade
+
+import (
+	"os"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestUpgradeAToB(t *testing.T) {
+	versionA := os.Getenv("FLEET_VERSION_A")
+	if versionA == "" {
+		t.Skip("Missing environment variable FLEET_VERSION_A")
+	}
+
+	versionB := os.Getenv("FLEET_VERSION_B")
+	if versionB == "" {
+		t.Skip("Missing environment variable FLEET_VERSION_B")
+	}
+
+	f := NewFleet(t, versionA)
+
+	client, err := f.Client()
+	require.NoError(t, err)
+
+	// enroll a host
+	hostname, err := f.StartHost()
+	require.NoError(t, err)
+
+	// wait until host is enrolled and software is listed
+	require.Eventually(t, func() bool {
```

Comment:
nit: Extract this statement into a function to remove duplication

Reply:
done